### DISCUSSION TITLE
Rendering Discrepancy on Older Android Smartphones (e.g., LG G6)

### DISCUSSION THREAD

**Discussion by dsilvera** [sceneview/sceneview#397] [General]
**Issue Description**
I am facing a rendering issue with the SceneView library on certain older Android smartphones, specifically observed on an LG G6 running Android 9. The problem manifests as an excessively elevated perception of light, resulting in an inconsistent rendering of my 3D model. This issue is not present on newer smartphones.

**Code Snippet for light**
```
environment = environmentLoader.createHDREnvironment(hdrFilePath)!!
        environment.indirectLight?.intensity = 60_000F
        val colorGrading: ColorGrading = ColorGrading.Builder()
            .toneMapper(ToneMapper.Linear())
            .contrast(0.955f)
            .vibrance(0.952f)
            .exposure(0.0835f)
            .saturation(0.867f)
            .build(engine)

        view.colorGrading = colorGrading

        val (r, g, b) = Color(0.9098f, 0.5294f, 0.2627f, 1.0f) //Colors.cct(6_500.0f)

        lightNodeAmbien = LightNode(engine, LightManager.Type.DIRECTIONAL) {
            color(r, g, b)
            intensity(180_100.0f) //320_000.0f is good
            position(0f, 0f, 0f)
            direction(0f, 0f, -10f)
            castShadows(false)
        }

        addChildNode(lightNodeAmbien!!)
        cameraNode.addChildNode(lightNodeAmbien!!)

        val (r2, g2, b2) = Color(0.6745f, 0.7059f, 0.9098f, 1.0f)

        lightNode1 = LightNode(engine, LightManager.Type.DIRECTIONAL) {
            color(r2, g2, b2)
            intensity(180_300.0f)
            direction(-0.55f, -0.15f, 0.0f)
            castShadows(false)
        }

        addChildNode(lightNode1!!)
        cameraNode.addChildNode(lightNode1!!)

        val (r3, g3, b3) = Color(0.9804f, 0.7333f, 0.7255f, 1.0f)

        lightNode2 = LightNode(engine, LightManager.Type.POINT) {
            color(r3, g3, b3)
            intensity(60_000.0f)
            position(0.05f, -0.055f, 0.03f)
            castShadows(false)
        }

        addChildNode(lightNode2!!)
        cameraNode.addChildNode(lightNode2!!)
```

**Problem Details**
Upon closer inspection, I have noticed that the rendering inconsistency is more pronounced on devices with outdated specifications, such as the LG G6. The issue persists despite employing a consistent codebase and rendering parameters across various devices.

**Additional Context**
Affected Device: LG G6
Android Version: 9

**Screenshots**
![Screenshot_20240118-155312](https://github.com/SceneView/sceneview-android/assets/1682961/25d54095-3f71-40d7-8b6f-1a67058cf166)
![Screenshot_20240118-160138](https://github.com/SceneView/sceneview-android/assets/1682961/345a0ca4-e04c-4402-a782-9410178a5e0b)

**Comment by grassydragon** [comment]
Hi!
Since you're saying that the code is the same for all devices, there can be a rendering issue related to the Filament engine used in Sceneview. Additionally, I haven't found any condition in Sceneview that can change rendering parameters on Android 9. However, you can try disabling some lighting effects (for example, ambient occlusion) to see if it helps.

**Reply by dsilvera** [reply]
Thank you for your reply.

I found the solution. If I set the parameter indirectLightSpecularFilter to false when I created the environment, it works for all devices :

```
environment = environmentLoader.createHDREnvironment(
            hdrFilePath,
            indirectLightSpecularFilter = false
        )
```

**Comment by ThomasGorisse** [comment]
Rendering differences on older devices are usually caused by limited OpenGL ES support or GPU driver bugs. Filament (SceneView's renderer) requires at least OpenGL ES 3.0.

v4.0.0-rc.1 includes Filament 1.70.2 with improved compatibility for older GPUs. If you still see discrepancies:
1. Check that the device supports OpenGL ES 3.0+
2. Simplify materials (avoid complex PBR on low-end devices)
3. Reduce texture resolution for older hardware
4. Test with the Filament sample viewer to isolate SceneView vs Filament-level issues
